aws/resource_aws_glue_catalog_database.go

```diff
@@ -0,0 +1,180 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsGlueCatalogDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlueCatalogDatabaseCreate,
+		Read:   resourceAwsGlueCatalogDatabaseRead,
+		Update: resourceAwsGlueCatalogDatabaseUpdate,
+		Delete: resourceAwsGlueCatalogDatabaseDelete,
+		Exists: resourceAwsGlueCatalogDatabaseExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"catalog_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"location_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"parameters": {
+				Type:     schema.TypeMap,
+				Elem:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsGlueCatalogDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
+	glueconn := meta.(*AWSClient).glueconn
+	catalogID := createAwsGlueCatalogID(d, meta)
+	name := d.Get("name").(string)
+
+	input := &glue.CreateDatabaseInput{
+		CatalogId: aws.String(catalogID),
+		DatabaseInput: &glue.DatabaseInput{
+			Name: aws.String(name),
+		},
+	}
+
+	_, err := glueconn.CreateDatabase(input)
+	if err != nil {
+		return fmt.Errorf("Error creating Catalogue Database: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", catalogID, name))
+
+	return resourceAwsGlueCatalogDatabaseUpdate(d, meta)
+}
+
+func resourceAwsGlueCatalogDatabaseUpdate(d *schema.ResourceData, meta interface{}) error {
+	glueconn := meta.(*AWSClient).glueconn
+
+	catalogID, name := readAwsGlueCatalogID(d.Id())
+	input := &glue.UpdateDatabaseInput{
+		CatalogId: aws.String(catalogID),
+		DatabaseInput: &glue.DatabaseInput{
+			Name:        aws.String(name),
+			Description: aws.String(d.Get("description").(string)),
+			LocationUri: aws.String(d.Get("location_uri").(string)),
+			Parameters:  make(map[string]*string),
+		},
+		Name: aws.String(name),
+	}
+	for key, value := range d.Get("parameters").(map[string]interface{}) {
+		input.DatabaseInput.Parameters[key] = aws.String(value.(string))
+	}
+
+	doUpdate := d.HasChange("description")
+	doUpdate = doUpdate || d.HasChange("location_uri")
+	doUpdate = doUpdate || d.HasChange("parameters")
+
+	if doUpdate {
```

Comment:
Nitpick, but this is a tiny little bit more difficult to read/understand - could be simplified to `if d.HasChange("description") || d.HasChange("location_uri") || d.HasChange("parameters") {`.